tests/test_client.py

```diff
@@ -0,0 +1,84 @@
+from textwrap import dedent
+from typing import Dict
+
+from git_machete.client import MacheteClient
+from git_machete.git_operations import GitContext, LocalBranchShortName
+from git_machete.utils import Annotation
+from .mockers import (GitRepositorySandbox, rewrite_definition_file)
+
+
+class TestClient:
+
+    def setup_method(self) -> None:
+
+        self.repo_sandbox = GitRepositorySandbox()
+
+        (
+            self.repo_sandbox
+            # Create the remote and sandbox repos, chdir into sandbox repo
+            .new_repo(self.repo_sandbox.remote_path, "--bare")
+            .new_repo(self.repo_sandbox.local_path)
+            .execute(f"git remote add origin {self.repo_sandbox.remote_path}")
+            .execute('git config user.email "tester@test.com"')
+            .execute('git config user.name "Tester Test"')
+        )
+
+    def test_annotations_read_definition_file(self) -> None:
+        """
+        Verify behaviour of a 'MacheteClient.read_definition_file()' method
+        """
+        (
+            self.repo_sandbox
+                .new_branch('master')
+                .commit()
+                .push()
+                .new_branch('feature1')
+                .commit()
+                .new_branch('feature2')
+                .commit()
+                .check_out("master")
+                .new_branch('feature3')
+                .commit()
+                .push()
+                .new_branch('feature4')
+                .commit()
+                .check_out("master")
+                .new_branch('feature5')
+                .commit()
+                .new_branch('feature6')
+                .commit()
+                .check_out("master")
+                .new_branch('feature7')
+                .push()
+        )
+
+        body: str = \
+            """
+            master
+            feature1
+            feature2 annotation
+            feature3 annotation rebase=yes push=yes
```

Comment:
TBD: if both `rebase=no` and `rebase=yes` occur at the same time.

In fact... I'd reconsider if `...=yes` qualifiers should even be supported at all 🤔

Reply:
>   TBD: if both rebase=no and rebase=yes occur at the same time.

1️⃣ @PawelLipski I was thinking about adding some check which won't allow to have both `rebase=no` and `rebase=yes` qualifiers at the same time.

>   In fact... I'd reconsider if ...=yes qualifiers should even be supported at all

2️⃣ Well, traverse does rebase and push by default so indeed there is no need to have ...=yes qualifiers, so we agree that only `=no` qualifiers are supported?

Reply:
Indeed, let's only support `...=no`!